motor_control/throttle/base_throttle.ino

```diff
@@ -1,15 +1,54 @@
 #include "base_throttle.h"
 
-Throttle::Throttle(const int &pin)
+Throttle::Throttle()
 {
-    m_sensor_pin = pin;
-    m_raw_value = 0;
-    m_motor_value = 0;
+    lastMotorWrite = 0;
+}
+
+
+void Throttle::run(){
+    setup();
+    loop();
+}
+
+void Throttle::setup() {
+  // Initialisation of ESC
+  ESC.attach(ESCOutput, minPulseWidth, maxPulseWidth);
+}
+
+void Throttle::writeToESC(int input_pin) { 
+  ESC.write(map(input_pin, 0, 1023, 0, 180));
+}
+
+int Throttle::readThrottle() {
+  /*
+   * Reads the throttle input from analog signal,
+   * Returns 0~1023
+   */
+  int throttleReading = analogRead(throttleIn);
+  int throttleMin = throttleMinVoltage * 1023; 
+  int throttleMax = throttleMaxVoltage * 1023;
+  // Throttle cutoff threshold
+  if (throttleReading < throttleMin) {
+    return 0;
+  } else if (throttleReading < throttleMax) {
+    return map(throttleReading, throttleMin, throttleMax, 0, 1023);
+  } else {
+    return 1023;
+  }
```

Comment:
By now from the comments above, it should be clear that there are a lot of changes to be made here. Please drop a reply here if you're not sure what those are though!